pkg/controller/direct/monitoring/monitoringdashboard_controller.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+
+	api "cloud.google.com/go/monitoring/dashboard/apiv1"
+	pb "cloud.google.com/go/monitoring/dashboard/apiv1/dashboardpb"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/monitoring/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/directbase"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/references"
+)
+
+func init() {
+	directbase.ControllerBuilder.RegisterModel(krm.MonitoringDashboardGVK, newDashboardModel)
+}
+
+func newDashboardModel(config *controller.Config) directbase.Model {
+	ctx := context.TODO()
+	gcpClient, err := newGCPClient(ctx, config)
```

Comment:
or we can move this to `AdapterForObject` so as we can use the passed in `context.Context` parameter.
The AdapterForObject will be initialized in each Reconcile loop.

Reply:
My one concern is that building the client may be expensive: if we have to open a new TCP connection, or if we have to get the token again.  But I don't want to block this PR on it, so I'll move it!

Reply:
Ah, I guess we're calling the expensive call every time anyway :-)